### PR TITLE
feat(config-validator): add --strict option

### LINKED ISSUE #22026: Add strict mode to config validator
### What would you like Renovate to be able to do?

Add `--strict` flag to our Renovate config validator so it throws an error when a scanned Renovate config needs to be migrated.

### If you have any ideas on how this should be implemented, please tell us here.

We probably need to change some code here: https://github.com/renovatebot/renovate/blob/main/lib/config-validator.ts

### Is this a feature you are interested in implementing yourself?

No

### PR DESCRIPTION
**Pull request by fgreinacher**
## Changes

Adds a `--strict` option to `renovate-config-validator` that makes it fail if config migration is needed.

## Context

Closes #22026

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

🛠 with ❤️ by [Siemens](https://opensource.siemens.com/)

### CODE REVIEW

**Comment by fgreinacher** [comment]
Tested manually in several variations:

```terminal
$ yarn run config-validator --strict
yarn run v1.22.19
$ ts-node lib/config-validator.ts --strict
 INFO: Validating renovate.json
 WARN: Config migration necessary
       "oldConfig": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "baseBranch": "main"
       },
       "newConfig": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "baseBranches": ["main"]
       }
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
$ yarn run config-validator --strict renovate.json
yarn run v1.22.19
$ ts-node lib/config-validator.ts --strict renovate.json
 INFO: Validating renovate.json
 WARN: Config migration necessary
       "oldConfig": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "baseBranch": "main"
       },
       "newConfig": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "baseBranches": ["main"]
       }
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
$ yarn run config-validator renovate.json --strict
yarn run v1.22.19
$ ts-node lib/config-validator.ts renovate.json --strict
 INFO: Validating renovate.json
 WARN: Config migration necessary
       "oldConfig": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "baseBranch": "main"
       },
       "newConfig": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "baseBranches": ["main"]
       }
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
$ yarn run config-validator --strict renovate.json renovate_2.json 
yarn run v1.22.19
$ ts-node lib/config-validator.ts --strict renovate.json renovate_2.json
 INFO: Validating renovate.json
 WARN: Config migration necessary
       "oldConfig": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "baseBranch": "main"
       },
       "newConfig": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "baseBranches": ["main"]
       }
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
$ yarn run config-validator renovate.json renovate_2.json       
yarn run v1.22.19
$ ts-node lib/config-validator.ts renovate.json renovate_2.json
 INFO: Validating renovate.json
 WARN: Config migration necessary
       "oldConfig": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "baseBranch": "main"
       },
       "newConfig": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "baseBranches": ["main"]
       }
 INFO: Validating renovate_2.json
 WARN: Config migration necessary
       "oldConfig": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "baseBranch": "main"
       },
       "newConfig": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "baseBranches": ["main"]
       }
 INFO: Config validated successfully
✨  Done in 4.35s.
```

File: renovate.json [hunk 1]
```diff
@@ -1,42 +1,4 @@
...
-  "regexManagers": [
-    {
-      "fileMatch": ["config/options/index.ts"],
-      "matchStrings": [
-        "default: '(?<depName>ghcr.io/containerbase/sidecar):(?<currentValue>.*)'"
-      ],
-      "datasourceTemplate": "docker"
-    }
-  ]
+  "baseBranch": "main"
```
Comment (viceice) [inline renovate.json:3]: revert

File: renovate_2.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (viceice) [inline renovate_2.json:1]: remove file

**Review by viceice — CHANGES_REQUESTED** [review]

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Comment by viceice** [comment]
btw: We should use commander for arg parsing on validator too 🙃 makes it easier to add future flags

**Pull request merged**